### PR TITLE
Category filter form added to checklist index, with option to edit categories in the admin.

### PR DESCRIPTION
**Pull request by Domchap90**
# Forecast Ticket Link(s)

https://app.forecast.it/project/P-155/task-board/T26512

### CODE REVIEW

**Review by jakequirke — APPROVED** [review]

File: checklists/forms.py [hunk 1]
```diff
@@ -17,4 +17,20 @@ class ChecklistForm(forms.ModelForm):
...
+
+class ChecklistCategoryFilterForm(forms.Form):
+
+    category = forms.ModelChoiceField(
+        queryset=Category.objects.all(), empty_label="--No category selected--"
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self
```
Comment (Will-Hoey) [inline checklists/forms.py:31]: What's this?

File: checklists/migrations/0008_checklist_archived.py
Comment (Will-Hoey) [inline checklists/migrations/0008_checklist_archived.py:1]: How come this was deleted?

File: checklists/models.py [hunk 2]
```diff
@@ -7,6 +7,17 @@
...
 
+class Category(models.Model):
+    name = models.CharField(max_length=255)
+    slug = models.SlugField(unique=True)
+
+    class Meta:
+        verbose_name_plural = "Categories"
+
+    def __str__(self):
+        return f"{self.name}"
```
Comment (Will-Hoey) [inline checklists/models.py:18]: Not big but given name is already a string you don't need to make this an f string

File: checklists/migrations/0008_checklist_archived.py
Comment (Domchap90) [reply checklists/migrations/0008_checklist_archived.py:1]: Was conflicting with an existing migration.

**Pull request merged**